monitoring/mock_uss/interaction_logging/logger.py

```diff
@@ -47,41 +37,34 @@ def log_file(code: str, content: Interaction) -> str:
     logname = "{}.json".format(basename)
     fullname = os.path.join(log_path, logname)
 
-    dump = copy.deepcopy(content)
     with open(fullname, "w") as f:
-        json.dump(dump, f, indent=2)
+        json.dump(content, f)
 
 
-def log_flask_interaction(request: Request, response: Response):
-    """
-    Logs flask request and response in an interaction
-    Args:
-        function
+class InteractionLoggingHook(QueryHook):
+    def on_query(self, query: Query) -> None:
+        # TODO: Make this configurable instead of hardcoding exactly these query types
+        if "query_type" in query and query.query_type in {
+            QueryType.F3548v21USSGetOperationalIntentDetails,
+            QueryType.F3548v21USSNotifyOperationalIntentDetailsChanged,
+        }:
+            log_interaction(QueryDirection.Outgoing, query)
 
-    Returns:
-        function response
-    """
-    method = request.method
-    url = request.url
-    if "/uss/v1/" not in url:
-        return
-    type = ""
-    if "telemetry" in url:
-        type = "Pos"
-    elif "operational_intent" in url:
-        type = "Op"
-    elif "constraint" in url:
-        type = "Constr"
-    else:
-        type = "Unknown"
-    st = datetime.datetime.utcnow()
-    rt = (datetime.datetime.utcnow() - st).total_seconds()
-    logger.debug(f"res - {str(response)}")
-    query = describe_flask_query(request, response, rt)
-    issues = []
-    if query.status_code != 200 or query.status_code != 204:
-        issue = Issue(description=response.get_data(as_text=True))
-        issues.append(issue)
-    interaction = Interaction(query=query, reported_issues=issues)
-    log_file(f"incoming_{method}_{type}", interaction)
+
+query_hooks.append(InteractionLoggingHook())
+
+
+# https://stackoverflow.com/a/67856316
+@webapp.before_request
+def interaction_log_before_request():
+    flask.Flask.custom_profiler = {"start": datetime.datetime.utcnow()}
+
+
+@webapp.after_request
+def interaction_log_after_request(response):
+    elapsed_s = (
+        datetime.datetime.utcnow() - flask.current_app.custom_profiler["start"]
+    ).total_seconds()
+    query = describe_flask_query(flask.request, response, elapsed_s)
+    log_interaction(QueryDirection.Incoming, query)
```

Comment:
This will log the qualifier interface injection endpoint queries also. Whereas, we want to log only the interuss requests. Hence, we should add logic to skip logging if  "/uss/v1/" not in url.

Reply:
Added hardcoded filter + TODO to make it configurable.